src/callbacks_step/timeseries_dg2d.jl

```diff
@@ -0,0 +1,139 @@
+
+# Creates cache for time series callback
+function create_cache_timeseries(point_coordinates, mesh::TreeMesh{2}, dg, cache)
+  # Determine element ids for point coordinates
+  element_ids = get_elements_by_coordinates(point_coordinates, mesh, dg, cache)
+
+  # Calculate & store Lagrange interpolation polynomials
+  interpolating_polynomials = calc_interpolating_polynomials(point_coordinates, element_ids, mesh,
+                                                             dg, cache)
+
+  timeseries_cache = (; element_ids, interpolating_polynomials)
+
+  return timeseries_cache
+end
+
+
+# Find element ids containing coordinates given as a matrix [ndims, npoints]
+function get_elements_by_coordinates!(element_ids, coordinates, mesh::TreeMesh, dg, cache)
+  @assert length(element_ids) == size(coordinates, 2) "size mismatch"
```

Comment:
`@assert` might be disabled on certain optimization levels in the future. I think you want to throw a `DimensionMismatch` error here.